### PR TITLE
Predication of non-divisible splits

### PR DESCRIPTION
**Pull request by naoyam**
(Stacked on #2677)

Basically just porting what's already done with the current predication

### CODE REVIEW

**Comment by naoyam** [comment]
!build

File: csrc/id_model/indexing.cpp [hunk 1]
```diff
@@ -1086,6 +1089,41 @@ std::vector<PredicateInfo> TensorIndexer::getPredicates(
     info_vec.emplace_back(info);
   }
 
+  // Add predicates for non-divisible splits.
+  // If this is a reduction init expr, then no need to take care of
+  // non divisible splits
+  if (!lower_utils::isReductionInitExpr(expr)) {
+    // for (const auto& [eg, direction] :
+    // non_divisible_split_predicates) {
```
Comment (jacobhinkle) [inline csrc/id_model/indexing.cpp:1097]:
```suggestion
```

**Comment by naoyam** [comment]
Pinging @jacobhinkle

**Review by jacobhinkle — APPROVED** [review]
LGTM

**Pull request merged**